indico/modules/events/abstracts/legacy.py

```diff
@@ -0,0 +1,498 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from datetime import timedelta
+
+from indico.core import signals
+from indico.core.db import db
+from indico.core.db.sqlalchemy.util.session import no_autoflush
+from indico.modules.events.abstracts.models.abstracts import Abstract
+from indico.modules.events.abstracts.models.fields import AbstractFieldValue
+from indico.modules.events.abstracts.models.judgments import Judgment
+from indico.modules.events.abstracts.settings import abstracts_settings
+from indico.modules.events.contributions.models.fields import ContributionField
+from indico.modules.events.contributions.operations import create_contribution
+from indico.modules.events.models.persons import EventPerson
+from indico.modules.events.contributions.models.persons import AuthorType, ContributionPersonLink
+from indico.modules.users.models.users import User, UserTitle
+from indico.util.caching import memoize_request
+from indico.util.i18n import _
+from indico.util.string import encode_utf8, to_unicode
+from indico.util.text import wordsCounter
+
+FIELD_TYPE_MAP = {
+    'input': 'text',
+    'textarea': 'text',
+    'selection': 'single_choice'
+}
+
+
+@signals.event.contributions.contribution_deleted.connect
+def _contribution_deleted(contrib, parent=None):
+    from MaKaC.review import AbstractStatusAccepted, AbstractStatusSubmitted
+    abstract = contrib.abstract
+    if abstract:
+        status = abstract.as_legacy.getCurrentStatus()
+        if isinstance(status, AbstractStatusAccepted):
+            if status.getTrack() is not None:
+                abstract.as_legacy.addTrack(status.getTrack())
+            abstract.as_legacy.setCurrentStatus(AbstractStatusSubmitted(abstract.as_legacy))
+            abstract.contribution = None
+
+
+def person_from_data(person_data, event):
+    user = User.find_first(~User.is_deleted, User.all_emails.contains(person_data['email'].lower()))
+    if user:
+        return EventPerson.for_user(user, event)
+
+    person = EventPerson.find_first(email=person_data['email'].lower())
```

Comment:
:warning: This finds event persons from any event, not just from the event you care about! `event.persons.filter_by(email=...).first()